README.md

```diff
@@ -1,97 +1,146 @@
-# Overview
+# Overview #
 
 _node-zookeeper - A Node.js client for Apache Zookeeper._
 
 This module is implemented on top of the ZooKeeper C API; consult the [ZK Reference](http://zookeeper.apache.org/doc/r3.4.13/index.html) for further details on behavior.
 
 __Latest changes__ are described in the [changelog](./CHANGELOG.md)
 
-## Installation
+## Installation ##
 (note the name `zookeeper` in lowercase)
 ```bash
 npm install zookeeper
 ```
 
-# Example
+# Examples #
 
+Use the `async/await` enabled client:
 ```javascript
-const ZooKeeper = require('zookeeper');
+const ZooKeeper = require('zookeeper').Promise;
+```
+Note: checkout the API for the [async/await enabled client here](#methods-asyncawait-enabled-client)
 
-function onCreate(client, rc, error, path) {
-    if (rc !== 0) {
-        console.log(`zk node create result: ${rc}, error: '${error}', path=${path}`);
-    } else {
-        console.log(`created zk node ${path}`);
+There is also a callbacks based client:
+```javascript
+const ZooKeeper = require('zookeeper');
+```
+Note: checkout the API for the [callback based client here](#methods-callbacks-based-client)
 
-        process.nextTick(() => {
-            client.close();
-        });
-    }
+#### Example: create a client ####
+Create an instance of the ZooKeeper client using `async/await`:
+```javascript
+function createClient(timeoutMs = 5000) {
+    const config = {
+        connect: host,
+        timeout: timeoutMs,
+        debug_level: ZooKeeper.ZOO_LOG_LEVEL_WARN,
+        host_order_deterministic: false,
+    };
+
+    return new ZooKeeper(config);
 }
 
-function onConnect(client, err) {
-    if (err) {
-        throw err;
-    }
+const client = createClient();
+```
+The client is ready when connected to a ZooKeeper server:
+```javascript
+client.init(config);
 
-    console.log(`zk session established, id=${client.client_id}`);
-    client.a_create('/node.js1', 'some value', ZooKeeper.ZOO_SEQUENCE | ZooKeeper.ZOO_EPHEMERAL, onCreate.bind(null, client));
-}
+await client.on_connected();
```

Comment:
I'm having second thoughts about this one. The function will probably cause confusion and likely bugs too. I'll remove it from the docs, and mark the function as deprecated.

Motivation: the `on_connected` function could potentially miss the connect event, if executed after init. If done before, `await` cannot be used.